[CUMULUS-2368] Add QueueWorkflow task

dopeters opened the pull request:
**Summary:** Develop QueueWorkflow task

Addresses [CUMULUS-2368: Develop QueueWorkflow task](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2368)

## Changes

* Added a QueueWorkflow task
* It takes the following inputs:
	- `workflow` - the workflow to be queued
	- `workflowInput` - the input for the queued workflow
	- `queueUrl` - (optional) url to specify a queue

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [ ] Integration tests

Jkovarik reviewed (changes requested):
Please add a task readme (e.g. tasks/move-granules/README)

flamingbear commented on tasks/queue-workflow/webpack.config.js:
```diff
@@ -0,0 +1,43 @@
+  optimization: {
+    nodeEnv: false
+  }
+};
```
Going to trust it is as correct as it is in the queue-granules and queue-pdr tasks

flamingbear reviewed (changes requested):
I have a couple of questions, and a few suggestions but it looks pretty good so far.

dopeters commented on packages/ingest/src/queue.js:
```diff
@@ -150,3 +150,61 @@ async function enqueueGranuleIngestMessage({
+  queueUrl,
+  parentWorkflow,
+  workflow,
+  workflowInput,
```
Done

dopeters commented on tasks/queue-workflow/tests/index.js:
```diff
@@ -0,0 +1,278 @@
+
+  t.is(messages.length, 1);
+
+  const receivedWorkflow = messages.map((message) => message.payload.workflow.name);
```
No, it's not! Changed

dopeters commented on tasks/queue-workflow/index.js:
```diff
@@ -0,0 +1,47 @@
+async function queueWorkflow(event) {
+  const workflow = event.input.workflow || {};
+  const workflowInput = event.input.workflowInput || {};
+  const arn = buildExecutionArn(
```
Borrowed this from the other tasks, but there's really no reason. Changed

dopeters commented:
@flamingbear Thanks for your thorough review here. I used a lot of the building blocks from the `queue-pdrs` and `queue-granules` tasks, so fresh eyes where really helpful for identifying places where things could be simplified/changed (especially in the tests). I think I addressed all of your comments, but let me know if you notice anything else.

Updated all tasks babel configs to use node 12.18.0
Also added a README as per @Jkovarik

flamingbear reviewed (approved):
Looks good, thanks for the updates. 👍

Jkovarik commented on tasks/queue-workflow/README.md:
```diff
@@ -0,0 +1,13 @@
+# @cumulus/queue-workflow
+
+Adds a workflow to a queue
```
Nit: Thanks for making the change to add this - I think we need to include a bit more context on the expected input/outputs for integrators, either here, the description strings or referenced to a doc, as we get support questions and/or have gotten feedback that we need more explanation/docs. 

A specific example of what I mean --  "what's a workflowInput", that can be as simple as 'the payload to the workflow to be executed' in the input schema and/or here.

dopeters replied:
Totally fair. Updated the README with all the param details

Jkovarik reviewed (approved):
Thanks for making those changes/addressing the doc concern!
